src/ComHelpers/ShapefileHelper.cpp

```diff
@@ -382,10 +382,88 @@ bool ShapefileHelper::GetClosestPoint(IShapefile* sf, double x, double y, double
             shp->Release();
         }
     }
-    // dist = minDist;
+    dist = minDist;
     return minDist < maxDistance;
 }
 
+// *****************************************************************
+//		GetClosestPoint()
+// *****************************************************************
+bool ShapefileHelper::GetClosestSnapPosition(IShapefile* sf, double x, double y, double maxDistance, std::vector<long>& ids,
+	long* shapeIndex, double& fx, double& fy, double& dist)
+{
+	if (!sf) return false;
+
+	VARIANT_BOOL vb;
+	double minDist = DBL_MAX;
+	for (long id : ids)
+	{
+		VARIANT_BOOL visible;
+		sf->get_ShapeVisible(id, &visible);
+		if (!visible) continue;

```

Comment:
It's best not to treat a VARIANT_BOOL as a bool.  If visible comes back as VARIANT_TRUE (-1), and you NOT that (!visible), the result is ambiguous.  It's better to say
if (visible == VARIANT_FALSE) continue;

Reply:
I'm pretty new to C and C++, most of the changes were made by looking at other code and copying what I need, thanks for the explanation. Can you point me to a good reference for this in particular?

Reply:
Here's a pretty good explanation, noting particularly to beware of comparing the various 'true' values:

[BOOL vs. VARIANT_BOOL vs. BOOLEAN vs. bool](https://blogs.msdn.microsoft.com/oldnewthing/20041222-00/?p=36923/)